src/GraphQL/Internal/Validation.hs

```diff
@@ -386,12 +472,77 @@ resolveFragmentDefinitions allFragments =
           modify (Set.insert name)
           FragmentSpread name directives <$> resolveFragment' definition
 
+-- * Variables
+
+data VariableDefinition = VariableDefinition Variable AST.Type (Maybe Value) deriving (Eq, Ord, Show)
```

Comment:
Done. I'm fond of record syntax in general, both for the extra documentation, and because there are times when accessors are really handy to have.